src/Platforms/SQLServerPlatform.php

```diff
@@ -1162,8 +1162,8 @@ public function getListNamespacesSQL()
             'SQLServerPlatform::getListNamespacesSQL() is deprecated,'
                 . ' use SQLServerSchemaManager::listSchemaNames() instead.',
         );
-
-        return "SELECT name FROM sys.schemas WHERE name NOT IN('guest', 'INFORMATION_SCHEMA', 'sys') AND name NOT IN(SELECT name FROM sys.database_principals WHERE type='R')";
+        // Exclude internal (<5) and backward compatibility existed schemas (>16383)
+        return "SELECT name FROM sys.schemas WHERE schema_id BETWEEN 5 AND 16383";
```

Comment:
Unless these changes can be integration-tested, please revert them. This method exists only for backward compatibility and should not be used. Those who are affected by this issue should migrate to the recommended API.